aws-cpp-sdk-transfer/include/aws/transfer/UploadFileRequest.h

```diff
@@ -83,6 +83,22 @@ class AWS_TRANSFER_API UploadFileRequest : public S3FileRequest, public std::ena
                       const std::shared_ptr<Aws::S3::S3Client>& s3Client, 
                       bool createBucket,
                       bool doConsistencyChecks);
+    UploadFileRequest(const Aws::String& fileName,
+                      const Aws::String& bucketName,
+                      const Aws::String& keyName,
+                      const Aws::String& contentType,
+                      const Aws::Map<Aws::String, Aws::String>& metadata,
+                      const std::shared_ptr<Aws::S3::S3Client>& s3Client,
+                      bool createBucket,
+                      bool doConsistencyChecks);
+    UploadFileRequest(const Aws::String& fileName,
+                      const Aws::String& bucketName,
+                      const Aws::String& keyName,
+                      const Aws::String& contentType,
+                      Aws::Map<Aws::String, Aws::String>&& metadata,
+                      const std::shared_ptr<Aws::S3::S3Client>& s3Client,
+                      bool createBucket,
+                      bool doConsistencyChecks);
     ~UploadFileRequest();
```

Comment:
This one is not your fault, I must have let it slip through a code-review internally, but the destructor needs to be virtual.